src/index.js

```diff
@@ -35,33 +33,26 @@ class SfrContentScript extends ContentScript {
     await this.waitForUserAuthentication()
   }
 
-  async waitForCurrentContract(contract) {
-    await waitFor(
-      async () => {
-        const el = document.querySelector(`li[id='${contract.id}']`)
-        if (el) {
-          el.click()
-          return false
-        } else {
-          const currentContract = await this.getCurrentContract()
-          if (!currentContract) {
-            return false
-          }
-          const result = currentContract.text === contract.text
-          return result
-        }
-      },
-      {
-        interval: 1000,
-        timeout: {
-          milliseconds: 10000,
-          message: new TimeoutError(
-            'waitForCurrentContract ' + contract.text + ' timed out after 10sec'
-          )
-        }
-      }
+  async navigateToNextContract(contract) {
+    this.log(
+      'info',
+      `📍️ navigateToNextContract starts for ${JSON.stringify(contract.text)}`
     )
-    return true
+    if (await this.isElementInWorker('#plusFac')) {
```

Comment:
I don't  understand why we have to remove some dom elements

Reply:
Cause sometimes I noticed the promise.race returns because we were finding one of the 3 awaited selector before actually changing the page. Removing them just ensure we're not finding them until the page  realoads 👍

Reply:
A small explanation on that in the code would be good, and after that approved